SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -684,21 +742,64 @@ bool RunSchemaCompiler()
 	}
 }
 
-bool SpatialGDKGenerateSchema()
+bool SpatialGDKGenerateSchema(bool bSaveSchemaDatabase, bool bRunSchemaCompiler)
 {
-	ResetUsedNames();
+	// Generate Schema for classes loaded in memory.
+
+	if (!SpatialGDKGenerateSchemaForClasses(GetAllSupportedClasses()))
+	{
+		return false;
+	}
+
+	if (bSaveSchemaDatabase)
+	{
+		if (!SaveSchemaDatabase())
```

Comment:
What's the validity of if `SpatialGDKGenerateSchemaForClasses` changes the in-memory schema database and we don't save it?

Reply:
Good question. I guess a restart would be recommended solution? It should be a pretty rare edge case since we check for write access before we do schema gen (in TryLoadExistingSchemaDatabase) so something funky would have to happen between then and this part.